src/fparser/__init__.py

```diff
@@ -65,11 +65,24 @@
 # First version by: Pearu Peterson <pearu@cens.ioc.ee>
 # First created: Oct 2006
 
+try:
+    from importlib.metadata import version, PackageNotFoundError
+except ModuleNotFoundError:
+    # Use backport package for python <3.8
+    from importlib_metadata import version, PackageNotFoundError
+try:
+    __version__ = version(__name__)
+except PackageNotFoundError:
+    from setuptools_scm import get_version
```

Comment:
I'm uncomfortable about the lack of test coverage here. What is it that requires that the package define a `__version__` and could it be implemented using a function (that can be tested) rather than just as code executed at import time?

Reply:
re testing, what I meant (but didn't manage to say) was that it would be better if this functionality (the `try...except` etc.) was encapsulated in a function, e.g. `_set_version`. We could then have `__version__ = _set_version()` but it would also be possible to have a test for the function to check that it works in the presence/absence of the various packages that it uses.
Alternatively, if it's not actually required then we could remove it altogether :-)